code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -365,37 +363,43 @@ public ProjectMember createProjectMember(Account project, Member member) {
     }
 
     @Override
-    public Account setDefaultProject(Account project, long accountId) {
-        if (project == null) {
-            throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, "Failed", "No project specified.", null);
-        }
-        Account account = getAccountById(accountId);
-        if (accountId != project.getId() && project.getKind().equalsIgnoreCase(ProjectConstants.TYPE)
-                && !account.getKind().equalsIgnoreCase(ProjectConstants.TYPE)) {
-            int updateCount = create()
-                    .update(ACCOUNT)
-                    .set(ACCOUNT.PROJECT_ID, project.getId())
-                    .where(ACCOUNT.ID
-                            .eq(accountId))
-                    .execute();
-
-            if (1 != updateCount) {
-                throw new ClientVisibleException(ResponseCodes.NOT_MODIFIED, "Failed", "Failed to update account default project.", null);
+    public void allProjectsHaveNonRancherIdMember(ExternalId externalId) {
+        //This operation is expensive if there are alot of projects and members however this is
+        //only called when auth is being turned on. In most cases this will only be called once.
+        Member newMember = new Member(externalId, "owner");
+        Set<ExternalId> externalIds = new HashSet<>();
+        externalIds.add(new ExternalId(String.valueOf(getAdminAccount().getId()), ProjectConstants.RANCHER_ID));
+        List<Account> allProjects = getAccessibleProjects(externalIds, true, null);
+        for(Account project: allProjects){
+            List<? extends ProjectMember> members = getActiveProjectMembers(project.getId());
+            boolean hasNonRancherMember = false;
+            for (ProjectMember member: members){
+                if (!member.getExternalIdType().equalsIgnoreCase(ProjectConstants.RANCHER_ID)){
+                    hasNonRancherMember = true;
+                    deactvateThenRemove(member);
+                }
+                if (hasNonRancherMember) break;
```

Comment:
how many non-rancher_id members can there be in a project?  did you intend to just remove the first one found?